sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobContainerAsyncClient.java

```diff
@@ -428,6 +428,62 @@ public String getEncryptionScope() {
             .map(response -> new SimpleResponse<>(response, null));
     }
 
+    /**
+     * Creates a new container within a storage account if it does not exist. For more information, see the
+     * <a href="https://docs.microsoft.com/rest/api/storageservices/create-container">Azure Docs</a>.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <!-- src_embed com.azure.storage.blob.BlobContainerAsyncClient.createIfNotExists -->
+     * <pre>
+     * client.createIfNotExists&#40;&#41;.subscribe&#40;
+     *     response -&gt; System.out.printf&#40;&quot;Create completed%n&quot;&#41;,
+     *     error -&gt; System.out.printf&#40;&quot;Error while creating container %s%n&quot;, error&#41;&#41;;
+     * </pre>
+     * <!-- end com.azure.storage.blob.BlobContainerAsyncClient.createIfNotExists -->
+     *
+     * @return A reactive response signalling completion, or null if the container already exists.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> createIfNotExists() {
```

Comment:
The min overloads should actually maybe be Mono<Boolean> so the customer knows whether or not it was created. Otherwise they can't tell from this when the async object completes if the operation happened or not (even though they'll be assured there was a container).

In that case, the docstring should read: @return A reactive response signaling completion. {@code True} indicates that a new container was created. {@code False} indicates a container already existed at this location.
@kasobol-msft Do you have an opinion on this?

Reply:
Yeah, it's a good idea. See https://stackoverflow.com/questions/59196530/why-reactor-monovoid-is-recognized-as-an-empty-mono .